examples/js/exporters/GLTFExporter.js

```diff
@@ -1028,6 +1004,74 @@ THREE.GLTFExporter.prototype = {
 
 			}
 
+			var forceIndices = options.forceIndices;
+			var isMultiMaterial = Array.isArray( mesh.material );
+
+			if ( ! forceIndices && isMultiMaterial ) {
+
+				// temporal workaround.
+				console.warn( 'THREE.GLTFExporter: Force index for a mesh with multi-material.', mesh );
+				forceIndices = true;
+
+			}
+
+			var forcedIndex = false;
+
+			if ( geometry.index === null && forceIndices ) {
+
+				var indices = new Uint32Array( geometry.attributes.position.count );
+
+				for ( var i = 0, il = indices.length; i < il; i ++ ) {
+
+					indices[ i ] = i;
+
+				}
+
+				geometry.index = new THREE.Uint32BufferAttribute( indices, 1 );
+
+				forcedIndex = true;
+
+			}
+
+			var materials = isMultiMaterial ? mesh.material : [ mesh.material ] ;
+			var groups = isMultiMaterial ? mesh.geometry.groups : [ { materialIndex: 0, start: undefined, count: undefined } ];
+
+			// assuming materials.length === groups.length
```

Comment:
Let's add some more detail in this comment, since it might not be obvious based on my spec questions below. Suggested:

> Export one primitive for each material on the mesh. Only a subset of cases are currently handled:
> * [×] Single material, no geometry groups
> * [×] N materials, N geometry groups each using full draw range
> * [  ] N materials, N geometry groups with different draw ranges
> * [  ] N materials, M > 1 geometry groups

^ If that looks right? I'm trying to list cases that are valid in three.js, but not everything that glTF might allow.

Reply:
Ah, this logic isn't correct. I should loop with `groups` and we don't need to assume `materials.length === groups.length`.

    for ( var i = 0, il = groups.length; i < il; i ++ ) {

I'll update.

Reply:
Updated, thanks.